### PR TITLE
HPCC-27667 Remove the (unused) concept of a component from jlog

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
Main motivation was to simplify the jlog interface.  I suspect more can be done, but this was the obvious first step.

File: system/jlib/jlog.cpp [hunk 1]
```diff
@@ -1836,73 +1686,34 @@ void CLogMsgManager::report(const LogMsgCategory & cat, const IException * excep
     pushMsg(new LogMsg(cat, getNextID(), unknownJob, exception->errorCode(), buff.str(), 0, port, session));
```
Comment (jakesmith) [inline system/jlib/jlog.cpp:1686]: >2022-05-20T13:38:38.2128830Z /Users/runner/work/HPCC-Platform/HPCC-Platform/system/jlib/jlog.cpp:1686:78: error: format string is not a string literal [-Werror,-Wformat-nonliteral]
2022-05-20T13:38:38.2228940Z     pushMsg(new LogMsg(cat, getNextID(), unknownJob, exception->errorCode(), buff.str(), 0, port, session));

looks like this is causing the compile error, but not sure why only in macos

File: dali/server/daserver.cpp [hunk 2]
```diff
@@ -75,9 +75,11 @@ MODULE_EXIT()
 
 void setMsgLevel(ILogMsgHandler * fileMsgHandler, unsigned level)
 {
+    /*
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply dali/server/daserver.cpp:78]: yes, I agree, I'd remove it. It has probably never been used, and isn't used in containerized mode.

File: system/jlib/jlog.cpp (hunk 1 above)
Comment (ghalliday) [reply system/jlib/jlog.cpp:1686]: It was a bug - I hadn't removed the 0 parameter.  Not sure why linux didn't catch it.

File: system/jlib/jlog.cpp [hunk 3]
```diff
@@ -1914,45 +1725,6 @@ void CLogMsgManager::report(const LogMsgCategory & cat, const LogMsgJobInfo & jo
     pushMsg(new LogMsg(cat, getNextID(), job, exception->errorCode(), buff.str(), 0, port, session));
```
Comment (jakesmith) [inline system/jlib/jlog.cpp:1725]: I think this line need changing to remove component (0) ?

**Review by jakesmith — COMMENTED** [review]
@ghalliday - looks like a very worthwhile clearout.

I think there's 1 more LogMsg constructor passing a component # that needs changing on line 1725:
    pushMsg(new LogMsg(cat, getNextID(), job, exception->errorCode(), buff.str(), 0, port, session));

**Review by jakesmith — APPROVED** [review]
@ghalliday - looks good

**Review by richardkchapman — APPROVED** [review]

**Comment by ghalliday** [comment]
@richardkchapman squashed and retargeted to 8.8.x (from master)

**Pull request merged**